src/test/libraryfilesynctest.cpp

```diff
@@ -0,0 +1,701 @@
+#include <gtest/gtest.h>
+
+#include <QFile>
+#include <QFileInfo>
+#include <QTemporaryDir>
+#include <memory>
+
+#include "library/library_prefs.h"
+#include "sources/metadatasource.h"
+#include "test/librarytest.h"
+#include "track/track.h"
+#include "util/fileinfo.h"
+
+namespace {
+
+const auto kTestDir = QDir(QDir::current().absoluteFilePath("src/test/id3-test-data"));
+
+const auto kTestFileWithMetadata = QFileInfo(kTestDir, QStringLiteral("cover-test-jpg.mp3"));
+
+const auto kTestFileWithoutMetadata = QFileInfo(kTestDir, QStringLiteral("empty.mp3"));
+
+enum class FileType {
+    WithMetadata,
+    WithoutMetadata,
+};
+
+void sleepBeforeUpdatingFileLastModifiedTime() {
```

Comment:
Functions should be named for what they do, not depending on the outer context. 
So this could be advanceFileTimeSleep() or such.

Reply:
The naming here should reveal where the function is ought to be used and has been chosen deliberately.

This is **NOT** a regular function as in regular code. It has been created for documentation purposes.